drivers/serial/Kconfig.nrfx

```diff
@@ -48,54 +49,108 @@ nrfx_uart_num = 0
 rsource "Kconfig.nrfx_uart_instance"
 endif
 
-if HAS_HW_NRF_UARTE1
 nrfx_uart_num = 1
+if HAS_HW_NRF_UARTE$(nrfx_uart_num)
 rsource "Kconfig.nrfx_uart_instance"
 endif
 
-if HAS_HW_NRF_UARTE2
 nrfx_uart_num = 2
+if HAS_HW_NRF_UARTE$(nrfx_uart_num)
 rsource "Kconfig.nrfx_uart_instance"
 endif
 
-if HAS_HW_NRF_UARTE3
 nrfx_uart_num = 3
+if HAS_HW_NRF_UARTE$(nrfx_uart_num)
+rsource "Kconfig.nrfx_uart_instance"
+endif
+
+nrfx_uart_num = 00
+if HAS_HW_NRF_UARTE$(nrfx_uart_num)
+rsource "Kconfig.nrfx_uart_instance"
+endif
+
+nrfx_uart_num = 20
+if HAS_HW_NRF_UARTE$(nrfx_uart_num)
+rsource "Kconfig.nrfx_uart_instance"
+endif
+
+nrfx_uart_num = 21
+if HAS_HW_NRF_UARTE$(nrfx_uart_num)
+rsource "Kconfig.nrfx_uart_instance"
+endif
+
+nrfx_uart_num = 22
+if HAS_HW_NRF_UARTE$(nrfx_uart_num)
+rsource "Kconfig.nrfx_uart_instance"
+endif
+
+nrfx_uart_num = 30
+if HAS_HW_NRF_UARTE$(nrfx_uart_num)
 rsource "Kconfig.nrfx_uart_instance"
 endif
 
 config NRFX_TIMER0
 	default y
-	depends on UART_0_NRF_HW_ASYNC_TIMER = 0 \
-		|| UART_1_NRF_HW_ASYNC_TIMER = 0 \
-		|| UART_2_NRF_HW_ASYNC_TIMER = 0 \
-		|| UART_3_NRF_HW_ASYNC_TIMER = 0
+	depends on UART_0_NRF_HW_ASYNC_TIMER = 0 || UART_1_NRF_HW_ASYNC_TIMER = 0 || \
+		   UART_2_NRF_HW_ASYNC_TIMER = 0 || UART_3_NRF_HW_ASYNC_TIMER = 0 || \
+		   UART_00_NRF_HW_ASYNC_TIMER = 0 || UART_20_NRF_HW_ASYNC_TIMER = 0 || \
+		   UART_21_NRF_HW_ASYNC_TIMER = 0 || UART_22_NRF_HW_ASYNC_TIMER = 0 || \
+		   UART_30_NRF_HW_ASYNC_TIMER = 0
```

Comment:
Changes like this one should be dropped as `HW_ASYNC` options concern the legacy shim only.

Reply:
You're right, removed.